### PR TITLE
Feat/orgunit migration

### PR DESCRIPTION
**Pull request by eperedo**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/86943uqk4

### :memo: Implementation

- [x] Migrate org unit (capture, output) to typescript and clean architecture
- [x] Add expiry account field to edit user form
- [x] Save locales in edit user form (interface and database language)
- [x] Add search org. unit option in table and edit form. (tei search org. unit)

### :video_camera: Screenshots/Screen capture

**Assign org. units**
[User-Extended-App_org_units.webm](https://github.com/EyeSeeTea/user-extended-app/assets/461124/b91317f7-fa7d-4325-96fb-456cb5649c39)

**Update account expiration date**
[User-Extended-account_expiry.webm](https://github.com/EyeSeeTea/user-extended-app/assets/461124/80028dae-97ee-4f4c-b8dc-876b40d43a75)

**Update user locales**
[User-Extended-App_locales.webm](https://github.com/EyeSeeTea/user-extended-app/assets/461124/21c7b7bc-08b9-493f-9e16-0806414942b4)

**Edit search org. unit in table**
[User-Extended-App_search_orgunit.webm](https://github.com/EyeSeeTea/user-extended-app/assets/461124/9825f975-80c1-45aa-81b4-0ea23949fd8b)

**Edit search org. unit in edit form**
[User-Extended-App_search_orgunit_edit_form.webm](https://github.com/EyeSeeTea/user-extended-app/assets/461124/b267b831-ee86-4843-bde5-d6de779be6a5)

**Edit search org. unit in bulk edit**
[User-Extended-App_search_orgunit_bulk_edit.webm](https://github.com/EyeSeeTea/user-extended-app/assets/461124/7dfced2f-5a32-47ce-aae0-8002b45ad5ed)

### :fire: Testing

#86943uqk4

### CODE REVIEW

File: src/webapp/hooks/useUser.ts
Comment (tokland) [inline src/webapp/hooks/useUser.ts:1]: [best practices] A file called `useUser` is expected to contain that hook. In this case, it has more than one. Then: `users.hooks.ts` (or similar)

**Review by tokland — COMMENTED** [review]
Code looking very good, just minor comment.

About the feature, it all seemed to work for me, except this scenario:

- Any row -> "Assign to search OUs"
- Select 1 org unit -> Save
- The assignation is persisted (I can see it in the rows and the Users app)
- But  execute again ""Assign to search OUs" -> no orgunits are selected. We should see the current selection.

**Review by tokland — COMMENTED** [review]
- [ ] Fix conflicts with base branch
- [ ] Testing with docker.eyeseetea.com/eyeseetea/dhis2-data:2.36.11.1-sp-ip-pro -> row user=amarvolo, "Assign to organisation units" -> select any OU -> save.  Snackbar: "internal error". On DHIS2 logs I see: `Invalid UUID string: (through reference chain: org.hisp.dhis.user.User["userCredentials"]->org.hisp.dhis.user.UserCredentials["createdBy"])`. That's because we are posting `createdBy:  {id: "", displayName: ""}`.

**Comment by eperedo** [comment]
> * [x]  Fix conflicts with base branch
> * [x]  Testing with docker.eyeseetea.com/eyeseetea/dhis2-data:2.36.11.1-sp-ip-pro -> row user=amarvolo, "Assign to organisation units" -> select any OU -> save.  Snackbar: "internal error". On DHIS2 logs I see: `Invalid UUID string: (through reference chain: org.hisp.dhis.user.User["userCredentials"]->org.hisp.dhis.user.UserCredentials["createdBy"])`. That's because we are posting `createdBy:  {id: "", displayName: ""}`.

@tokland in that server some users has a different id inside `userCredentials` which was generating an `internal server error`. To avoid this problem I'm always passing the id from the root level (see method `buildUsersToSave` in `UserD2ApiRepository.ts`)

**Review by tokland — APPROVED** [review]
Tested on 2.36.11 and 40.3.0. :+1:

**Pull request merged**